subsys/bluetooth/host/hci_core.c

```diff
@@ -3211,13 +3211,33 @@ static void le_ltk_neg_reply(u16_t handle)
 	bt_hci_cmd_send(BT_HCI_OP_LE_LTK_REQ_NEG_REPLY, buf);
 }
 
+static void le_ltk_reply(u16_t handle, u8_t *ltk)
+{
+	struct bt_hci_cp_le_ltk_req_reply *cp;
+	struct net_buf *buf;
+
+	buf = bt_hci_cmd_create(BT_HCI_OP_LE_LTK_REQ_REPLY,
+				sizeof(*cp));
+	if (!buf) {
+		BT_ERR("Out of command buffers");
+		return;
+	}
+
+	cp = net_buf_add(buf, sizeof(*cp));
+	cp->handle = handle;
+
+	/* use only enc_size bytes of key for encryption */
+	memcpy(cp->ltk, ltk, sizeof(cp->ltk));
```

Comment:
The code doesn't seem to match the comment above it. `sizeof(cp->ltk)` will always evaluate to 16, whereas the original code was taking `conn->le.keys->enc_size` into account. So this looks wrong to me.

Reply:
The comment is outdated, I will remove it.
Encryption key size masking should all be handled by `bt_smp_request_ltk`, it is correct to copy the entire array here, because we need to include the already zero-set bits.

Reply:
It's gone now.